### DISCUSSION TITLE
Production builds

### DISCUSSION THREAD

**Discussion by chrishare10** [onedarnleyroad/craftcms#51] [General]
Hey @johndwells, first off this is a fantastic setup and I’ve been using it for a few projects now. Thanks for putting it together.

I am confused however with regards to connecting assets in production. When I build the assets, is the assumption that I will link directly to the files in the dist/, or is there a step I’m missing in production to use the vite plugin and other functionality to wire up my frontend?

Thanks!

**Comment by johndwells** [comment]
@chrishare10 I'm glad you've gotten value out of this project!

> I am confused however with regards to connecting assets in production. When I build the assets, is the assumption that I will link directly to the files in the dist/, or is there a step I’m missing in production to use the vite plugin and other functionality to wire up my frontend?

If you're talking about creating `<link>` and `<script>` tags pointing to the compiled CSS & JS, the [Vite plugin](https://nystudio107.com/docs/vite/) handles all of this for you automatically. See [templates/_includes/scripts.twig](https://github.com/onedarnleyroad/craftcms/blob/master/templates/_includes/scripts.twig).

As for other static assets, such as SVGs or fonts, place those inside `src/public` and they will be copied over to `web/dist` when you run `make build` or `ddev exec npm run build`. You can learn more about this on [ViteJS's docs](https://vitejs.dev/guide/assets.html#the-public-directory).
